apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+        )}
+      </div>
+      <div className={styles.folderRowRight}>
+        <div className={styles.length}>&nbsp;</div>
+        {previewSound && (
+          <div className={styles.previewContainer}>
+            <FontAwesome
+              title={undefined}
+              icon={'play-circle'}
+              className={classNames(
+                styles.preview,
+                isPlayingPreview && styles.previewPlaying
+              )}
+              onClick={onPreviewClick}
+            />
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+// SoundPanelEntry: A single sound.
+
+interface SoundsPanelEntryProps {
+  currentValue: string;
+  playingPreview: string;
+  folder: SoundFolder;
+  sound: SoundData;
+  showingSoundsOnly: boolean;
+  onSelect: (path: string) => void;
+  onPreview: (path: string) => void;
+  currentSoundRefCallback?: (ref: HTMLDivElement) => void;
+}
+
+const SoundsPanelEntry: React.FunctionComponent<SoundsPanelEntryProps> = ({
+  currentValue,
+  playingPreview,
+  folder,
+  sound,
+  onSelect,
+  onPreview,
+  currentSoundRefCallback,
+}) => {
+  const soundPath = folder.id + '/' + sound.src;
+  const isSelected = soundPath === currentValue;
+  const isPlayingPreview = playingPreview === soundPath;
+
+  const onSoundSelect = useCallback(() => {
+    if (!isPlayingPreview) {
+      onPreview(soundPath);
+    }
+    onSelect(soundPath);
+  }, [isPlayingPreview, onPreview, onSelect, soundPath]);
+
+  const onSoundClick = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      e.stopPropagation();
+      onSoundSelect();
+    },
+    [onSoundSelect]
+  );
+
+  return (
+    <div
+      className={classNames(
+        'sounds-panel-sound-entry',
+        styles.soundEntry,
+        isSelected && styles.soundEntrySelected
+      )}
+      onClick={onSoundClick}
+      onKeyDown={event => {
+        if (event.key === 'Enter') {
+          onSoundSelect();
+        }
+      }}
+      ref={isSelected ? currentSoundRefCallback : null}
+      aria-label={sound.name}
+      tabIndex={0}
+      role="button"
+    >
+      <div className={styles.soundEntryMain}>
+        <div
+          className={classNames(
+            styles.iconContainer,
+            SoundStyle[sound.type]?.classNameBackground,
+            SoundStyle[sound.type]?.classNameBorder,
+            isSelected && styles.iconContainerSelected
+          )}
+        >
+          <FontAwesomeV6Icon
+            iconName={SoundStyle[sound.type]?.icon || ''}
+            className={classNames(
+              styles.typeIcon,
+              SoundStyle[sound.type]?.classNameColor
+            )}
+          />
+        </div>
+        <div
+          className={classNames(
+            styles.name,
+            sound.type === 'vocal' && styles.nameVocal
+          )}
+        >
+          {sound.name}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+// ScrollIntoView: This component does an eased scroll of the container's content,
```

Comment:
Very cool! This feels like something that should be in its own file for potential reuse (even if it stays within /music for now)

Reply:
Good call; done.